orbit-2-rxjava2/src/test/java/com/babylon/orbit2/RxJava2PluginDslBehaviourTest.kt

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright 2020 Babylon Partners Limited
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.babylon.orbit2
+
+import com.appmattus.kotlinfixture.kotlinFixture
+import io.reactivex.Completable
+import io.reactivex.Maybe
+import io.reactivex.Observable
+import io.reactivex.Single
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+internal class RxJava2PluginDslBehaviourTest {
+    private val fixture = kotlinFixture()
+    private val initialState = fixture<TestState>()
+
+    @BeforeEach
+    fun beforeEach() {
+        Orbit.registerDslPlugins(RxJava2Plugin)
+    }
+
+    @AfterEach
+    fun afterEach() {
+        Orbit.resetPlugins()
+    }
+
+    @Test
+    fun `single transformation flatmaps`() {
+        val action = fixture<Int>()
+
+        Middleware()
+            .given(initialState)
+            .whenever {
+                single(action)
+            }
+            .then {
+                states(
+                    { TestState(action + 5) }
+                )
+            }
+    }
+
+    @Test
+    fun `single transformation crashes if plugin is not included`() {
+        val action = fixture<Int>()
+        Orbit.resetPlugins()
```

Comment:
would this be a good place to use `@Nested` tests? two blocks, with plugin applied and without, having to reset it internally here feels a little messy